### PR TITLE
Blueprint start/stop listeners + ordering

### PR DESCRIPTION
**Pull request by channelcat**
Added blueprint start/stop listeners
Added sanic as the first argument in listeners triggered when the server starts/stops
reversed the order of response middleware

### CODE REVIEW

File: sanic/sanic.py [hunk 1]
```diff
@@ -71,7 +74,7 @@ def register_middleware(middleware):
             if attach_to == 'request':
                 self.request_middleware.append(middleware)
             if attach_to == 'response':
-                self.response_middleware.append(middleware)
+                self.response_middleware.insert(0, middleware)
```
Comment (jpiasetz) [inline sanic/sanic.py]: Why not use a [deque](https://docs.python.org/2/library/collections.html?highlight=collections#collections.deque)?

File: sanic/server.py [hunk 2]
```diff
@@ -157,7 +157,22 @@ def close_if_idle(self):
...
 
 
-def serve(host, port, request_handler, after_start=None, before_stop=None,
+def trigger_events(events, loop):
+    """
+    :param events: one or more sync or async functions to execute
+    :param loop: event loop
+    """
+    if events:
+        if type(events) is not list:
```
Comment (jpiasetz) [inline sanic/server.py]: isinstance of iterable might be better.

Comment (channelcat) [inline sanic/server.py]: Ah yes, will do :)

File: sanic/sanic.py (hunk 1 above)
Comment (jpiasetz) [inline sanic/sanic.py]: It's just an iterable. You can iterate over it as many times as you like provided you don't pop i.e.

```python
>>> from collections import deque
>>> my_list = deque(['b'])
>>> my_list.append('c')
>>> my_list.appendleft('a')
>>> for i in my_list:
...     print(i)
... 
a
b
c
>>> for i in my_list:
...     print(i)
... 
a
b
c
>>> 
```

Comment (channelcat) [inline sanic/sanic.py]: oh awesome, I did not know :)

Comment (channelcat) [inline sanic/sanic.py]: Added in https://github.com/channelcat/sanic/pull/105

**Pull request merged**